File: java/src/factorial.java
import java.util.*;
public class factorial {
    public static int CalculateFactorial(int x) {
        int fact=1;
        while(x>1){
        fact=fact*x;
        x=x-1;
        }
        return fact ;
    }

    public static void main(String args[]) {
        try (Scanner s = new Scanner(System.in)) {
            System.out.print("Enter number:");
            int num = s.nextInt();
            int fact =CalculateFactorial(num);
            System.out.print("factorial of "+num+" is : "+fact);
        }
    }
}

File: c/factorial.c
#include<stdio.h>
int factorial(int x){
    int output,n;
    printf("enter number:");
    scanf("%d",&n);
    if(n==0)
    printf("Not Possible");
else if (n==1)
printf("factorial of %d is 1",n);
else {
     output =  x * factorial(x-1);
}
    return output;
}
int main(){
    int result = factorial(n);
    printf("the factorial of %d is %d",n,result);
return 0;
}